lib/puppet/parser/resource.rb

```diff
@@ -171,11 +171,72 @@ def merge(resource)
     if evaluated?
       strict = Puppet[:strict]
       unless strict == :off
-        msg = _('Attempt to override an already evaluated resource%{location} with new values') %
-          { resource: ref, location: append_location(file, line, _(', defined'), ',') }
-        raise Puppet::ParseError.new(msg, resource.file, resource.line) if strict == :error
-        msg << append_location(resource.file, resource.line)
-        Puppet.warning(msg)
+        if strict == :error
```

Comment:
I do like the phrasing of these new messages better, so long as we're okay changing them (as in, we can assume no one is relying on their output for scripting). Seems fairly safe to me, especially if our tests passed.

Reply:
Given this, do we still need the `append_location` function?

Reply:
Actually the phrasing should be exactly the same as before.

In this case I'm wondering why if we have strict checking turned on you get a less helpful error message.... oh its because Puppet::ParseError adds its own way of saying where something is to the string.

  module ExternalFileError
....
    def to_s
      msg = super
      @file = nil if (@file.is_a?(String) && @file.empty?)
      if @file and @line and @pos
        "#{msg} at #{@file}:#{@line}:#{@pos}"
      elsif @file and @line
        "#{msg} at #{@file}:#{@line}"
      elsif @line and @pos
          "#{msg} at line #{@line}:#{@pos}"
      elsif @line
        "#{msg} at line #{@line}"
      elsif @file
        "#{msg} in #{@file}"
      else
        msg
      end

Of course these aren't translated so that should be awesome...

Reply:
I wanted to make sure things worked before I deleted append_location. I've got a build running without it.

Reply:
Okay, cool on it being the same string. Also 👍 on more clarity, seeing as apparently I couldn't even find where the "defined at" was coming from before xD